includes/Classifai/Providers/OpenAI/ChatGPT.php

```diff
@@ -379,6 +407,21 @@ public function setup_fields_sections() {
 			]
 		);
 
+		// Custom prompt for excerpt generation.
+		add_settings_field(
+			'generate_excerpt_prompt',
+			esc_html__( 'Prompt', 'classifai' ),
+			[ $this, 'render_textarea' ],
+			$this->get_option_name(),
+			$this->get_option_name() . '_excerpt',
+			[
+				'label_for'     => 'generate_excerpt_prompt',
+				'placeholder'   => $this->generate_excerpt_prompt,
+				'default_value' => $default_settings['generate_excerpt_prompt'],
+				'description'   => __( "Enter your custom prompt. If no custom prompt is entered, the default shown above will be used. Note the following variables that can be used in the prompt and will be replaced with content: {{WORDS}} - replaced with the excerpt length setting. {{TITLE}} will be replaced with the item's title.", 'classifai' ),
```

Comment:
Thinking about this some more, wondering if we really need the `{{WORDS}}` placeholder? `{{TITLE}}` makes sense as that will be different for each item you're generating an excerpt for but the actual length you're requesting is the exact same each time. Thinking if you're entering your own custom prompt, it's just as easy to put in the length value you want as it is to put in the `{{WORDS}}` placeholder.

So I'd suggest we remove that from this description (which also makes this description a little shorter which is nice) and then it would be great if on line 419, we take the excerpt prompt and replace `{{WORDS}}` in that with whatever value we have stored for the length, so the placeholder we render will show the proper amount of words instead of a placeholder

Reply:
Yes, agree. removing the `{{WORDS}}`  placeholder will simplify things. However, it will override the value of the "Excerpt length" field but most people won't go with a custom prompt So, it is fine. Committed your code suggestions.